src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx

```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
+
+```ts title="amplify/custom/LambdaInAVpcStack/resource.ts"
+import {
+  CfnOutput,
+  Duration,
+  Stack,
+  StackProps,
+  aws_ec2 as ec2,
+  aws_lambda as lambda,
+} from "aws-cdk-lib";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+import { Construct } from "constructs";
+import { dirname, join } from "path";
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+export class LambdaInAVpcStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    const { database } = props;
-
-    // create the backup plan
-    const plan = new backup.BackupPlan(this, 'BackupPlan', {
-      backupPlanRules: [
-        new backup.BackupPlanRule({
-          completionWindow: cdk.Duration.hours(2),
-          startWindow: cdk.Duration.hours(1),
-          scheduleExpression: events.Schedule.cron({
-            day: '15',
-            hour: '3',
-            minute: '30',
-          }),
-          moveToColdStorageAfter: cdk.Duration.days(30),
-        }),
+    const vpc = new ec2.Vpc(this, "LambdaVpc", {
+      subnetConfiguration: [
+        {
+          name: "Isolated",
+          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
+        },
+        {
+          name: "Private",
+          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+        },
+        {
+          name: "Public",
+          subnetType: ec2.SubnetType.PUBLIC,
+        },
       ],
     });
 
-    // add the database to the backup plan
-    plan.addSelection('Selection', {
-      resources: [backup.BackupResource.fromRdsDatabaseInstance(database)],
+    // Create a security group to be used on the lambda functions
+    const lambdaSecurityGroup = new ec2.SecurityGroup(
+      this,
+      "Lambda Security Group",
+      {
+        vpc,
+      }
+    );
+
+    const getDataLambda: NodejsFunction = new NodejsFunction(
+      this,
+      id + "-getDataLambda",
+      {
+        memorySize: 1024,
+        timeout: Duration.seconds(5),
+        runtime: lambda.Runtime.NODEJS_18_X,
+        handler: "handler",
+        entry: join(__dirname, "getData.ts"),
+        vpc: vpc,
+        vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_ISOLATED },
+        securityGroups: [lambdaSecurityGroup],
+      }
+    );
+
+    new CfnOutput(this, `${id}-getDataLambdaArn`, {
+      value: getDataLambda.functionArn,
+      exportName: `${id}-getDataLambdaArn`,
     });
   }
 }
 ```
 
+The Lambda function code for the `getDataLambda` is:
+
+```ts title="amplify/custom/LambdaInAVpcStack/getData.ts"
+import { APIGatewayProxyResultV2 } from "aws-lambda";
+
+// example data; example data can be fetched from a database
+const geoData = [
+  {
+    name: "United States",
+    states: [
+      "Alabama",
+      "Alaska",
+      "Arizona",
+      //...
+    ],
+  },
+];
+
+export async function handler(): Promise<APIGatewayProxyResultV2> {
```

Comment:
this signature is very different from the previous example, is this intentional? https://github.com/aws-amplify/docs/pull/6265/files#diff-41d0a6d7329d8852de45154989ef0af799ddbac5e7f825ec335e49fe5f31882eR132